src/map/utils/charutils.cpp

```diff
@@ -429,10 +433,15 @@ namespace charutils
             Sql_GetData(SqlHandle, 23, &campaign, &length);
             memcpy(&PChar->m_campaignLog, campaign, (length > sizeof(PChar->m_campaignLog) ? sizeof(PChar->m_campaignLog) : length));
 
-            PChar->SetPlayTime(Sql_GetUIntData(SqlHandle, 24));
-            PChar->profile.campaign_allegiance = (uint8)Sql_GetIntData(SqlHandle, 25);
-            PChar->setStyleLocked(Sql_GetIntData(SqlHandle, 26) == 1 ? true : false);
-            PChar->SetMoghancement(Sql_GetUIntData(SqlHandle, 27));
+            length = 0;
+            char* eminence = nullptr;
+            Sql_GetData(SqlHandle, 24, &eminence, &length);
+            memcpy(&PChar->m_eminenceLog, eminence, (length > sizeof(PChar->m_eminenceLog) ? sizeof(PChar->m_eminenceLog) : length));
```

Comment:
Thoughts on replacing memcpy and other C-style stuff with std::copy and the like?

Reply:
I am pro modern-style, in this instance since much of the code was mixing in with existing functions I wanted to maintain the same dialect since it was performing identical behaviour. I have long wanted to move areas like this to modern options, (ie. this here and especially config file handling .We're still using char*s and strcmp for crying out loud!)

Reply:
Also, as an aside, since these structs are necessarily comprised of only PoD types and no magic numbers are at play I'm less bothered by memcpy use in these cases.